crates/exex/src/execution.rs

```diff
@@ -121,6 +126,9 @@ impl<'a, DB: StateProvider> ShadowExecutor<'a, DB> {
                     },
                 };
 
+                // FIXME: why is result.logs always empty?
+                info!("Executed transaction: {:?}, {:?}", transaction.hash, result);
```

Comment:
It still seems like something is wrong here, but I'm still debugging. Executing with the diff here no longer panics, but it seems like we get mostly empty results back from revm. This line results in output like (note the empty logs arr on success results):

```
2024-06-26T13:00:48.168742Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Success { reason: Stop, gas_used: 23320, gas_refunded: 0, logs: [], output: Call(0x) }
2024-06-26T13:00:48.168819Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Success { reason: Stop, gas_used: 23320, gas_refunded: 0, logs: [], output: Call(0x) }
2024-06-26T13:00:48.168891Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Halt { reason: OutOfFunds, gas_used: 21000 }
2024-06-26T13:00:48.168964Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Halt { reason: OutOfFunds, gas_used: 21000 }
2024-06-26T13:00:48.169052Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Halt { reason: OutOfFunds, gas_used: 21000 }
2024-06-26T13:00:48.169124Z  INFO exex{id="ShadowExEx"}: shadow_reth_exex::execution: Executed transaction: 0x0000000000000000000000000000000000000000000000000000000000000000, Halt { reason: OutOfFunds, gas_used: 21000 }
```

Additionally, I'm not seeing shadow logs indexed in blocks where'd I'd expect to see them (e.g. post WETH deployment, using the default `shadow.json.example` included in the repo that shadows WETH). I think something still must be misconfigured with the revm config / env for these older blocks.

Reply:
It seems like this is an issue with the `HistoricalStateProvider`, running `transact` to have revm validate and it's pretty clear that we're getting bunk state (invalid nonces indicating account data isnt getting loaded into revm correctly):
```
2024-06-27T22:32:22.713599Z DEBUG exex{id="ShadowExEx"}: shadow_reth_exex::execution: Skipping invalid transaction err=nonce 7 too high, expected 0 transaction=TransactionSigned { hash: 0x0000000000000000000000000000000000000000000000000000000000000000, signature: Signature { r: 0x9e18efb7871f04f865e78affafef31e1238e7b9ec6688b280fdbaed20867a9a9_U256, s: 0x225f6b1ea4a5c0cc5174e81320a4b8720ca74e08767e3c95b5ea8df2a029970a_U256, odd_y_parity: true }, transaction: Legacy(TxLegacy { chain_id: Some(1), nonce: 7, gas_price: 4000000000, gas_limit: 60000, to: Call(0xb17e7e102e219450c2413a668e8f7ccad52deb3a), value: 0x0_U256, input: 0xa9059cbb000000000000000000000000540391f0ed1515a01d2fd6e7a59b1d824b9bf02c00000000000000000000000000000000000000000000000000000000ee6b2800 }) }
2024-06-27T22:32:22.713653Z DEBUG exex{id="ShadowExEx"}: shadow_reth_exex::execution: Skipping invalid transaction err=nonce 480629 too high, expected 0 transaction=TransactionSigned { hash: 0x0000000000000000000000000000000000000000000000000000000000000000, signature: Signature { r: 0xe15c8386edf46a56fb629187d5c672abd2575b3ac882ffac1254a6c1819a9d6a_U256, s: 0x4966ee50959a4139df2ff820e1ad75b201b5cb7cf15a6fe1eb0b7cde42963b3b_U256, odd_y_parity: false }, transaction: Legacy(TxLegacy { chain_id: None, nonce: 480629, gas_price: 4000000000, gas_limit: 36319, to: Call(0xc4131c1893576e078a0b637b653f3e6a18e137ac), value: 0x0_U256, input: 0xa9059cbb000000000000000000000000729343d5e8ced067f69b3986d1d3b82208cf5ff80000000000000000000000000000000000000000000000000000000000030d40 }) }
2024-06-27T22:32:22.713972Z DEBUG exex{id="ShadowExEx"}: shadow_reth_exex::execution: Skipping invalid transaction err=nonce 114 too high, expected 0 transaction=TransactionSigned { hash: 0x0000000000000000000000000000000000000000000000000000000000000000, signature: Signature { r: 0x6a0d7c25bcb28771ec6aa8acfda0632ee62f87dbfc70eb1bcb767d9f3f292392_U256, s: 0x03ccb206ebc7c25198696e39080485e3b98ae22f681e6941f04c30e1ac668d31_U256, odd_y_parity: false }, transaction: Legacy(TxLegacy { chain_id: Some(1), nonce: 114, gas_price: 4000000000, gas_limit: 21000, to: Call(0xb45ac4ba71c365b212f8f8ec1a2d115b5ea48295), value: 0x0000000000000000000000000000000000000000000000000029be6bf36e3100_U256, input: 0x }) }
```

I got a bit further by building the provider similar to [here](https://github.com/paradigmxyz/reth/pull/8092/files#diff-657bd8124fe7f214916a824250d154a60b579e12c0a03ff17ba715524bceb7d3R28-R35), but applying the n-1 bundle state output for each block in the chain. Unfortunately this seems pretty slow so far. It's also not clear to me why `chain.first().number.checked_sub(1)` is needed in the state provider there, as I think that'd put us a block behind where we'd want to be.